### PR TITLE
API Accounts/ProxyConfigs GET by host/version

### PR DESCRIPTION
**Pull request by Martouta**
Closes [THREESCALE-6310](https://issues.redhat.com/browse/THREESCALE-6310)
⚠️ Blocks [THREESCALE-6237](https://issues.redhat.com/browse/THREESCALE-6237)

image

image

Request: `curl -v  -X GET "http://provider-admin.3scale.localhost:3000/admin/api/account/proxy_configs/production.json?access_token=1&host=example.com&version=3"`

### CODE REVIEW

File: app/controllers/admin/api/account/proxy_configs_controller.rb [hunk 1]
```diff
@@ -33,15 +35,26 @@ def index
   private
 
   def proxy_configs
-    accessible_services_ids = System::Database.oracle? ? accessible_services.pluck(:id) : accessible_services.select(:id)
-    ProxyConfig.joins(:proxy).where(proxies: { service_id: accessible_services_ids }).by_environment(environment)
+    @proxy_configs ||= ProxyConfig.joins(:proxy)
+      .where(proxies: { service_id: accessible_services.pluck(:id) })
```
Comment (guicassolato) [inline app/controllers/admin/api/account/proxy_configs_controller.rb:39]:
```suggestion
      .where(proxies: { service_id: accessible_services.select(:id) })
```

Comment (Martouta) [reply app/controllers/admin/api/account/proxy_configs_controller.rb:39]: @guicassolato unfortunately that wouldn't work for all DBs. Anyway, my next PR is to improve the performance of all this and this part will change.

File: app/controllers/admin/api/account/proxy_configs_controller.rb [hunk 2]
```diff
@@ -25,6 +25,8 @@ class Admin::Api::Account::ProxyConfigsController < Admin::Api::BaseController
   #
   ##~ op.parameters.add @parameter_access_token
   ##~ op.parameters.add @parameter_environment
+  ##~ op.parameters.add :name => "host",    :description => "Find by host",    :dataType => "string", :required => false, :paramType => "query"
+  ##~ op.parameters.add :name => "version", :description => "Find by version", :dataType => "string", :required => false, :paramType => "query"
```
Comment (guicassolato) [inline app/controllers/admin/api/account/proxy_configs_controller.rb:29]: Should we describe them "Filter by..." instead of "Find by..."?

**Review by guicassolato — APPROVED** [review]

File: app/controllers/admin/api/account/proxy_configs_controller.rb (hunk 2 above)
Comment (Martouta) [reply app/controllers/admin/api/account/proxy_configs_controller.rb:29]: Makes sense 👍🏼 In the following PR I will add this improvement 👍🏼

**Pull request merged**